### PR TITLE
added /postvalidate route for JSON POST validation

### PR DESCRIPTION
**Pull request by calvinbutcher**
New "/postvalidate" route receives POST of JSON body-text with a Content-Type:application/json. Responds with JSON returning the received data, an okay flag, warnings, and errors.

### CODE REVIEW

File: implementations/validator/validator.py [hunk 1]
```diff
@@ -113,7 +136,8 @@ def do_test(self):
         return self.format_response(infojson, fmt)
 
     def dispatch_views(self):
-        self.app.route("/validate", "GET", self.do_test)
+        self.app.route("/validate", "GET", self.do_GET_test)
+        self.app.route("/postvalidate", "POST", self.do_POST_test)
```
Comment (azaroth42) [inline implementations/validator/validator.py]: What about just POSTing to /validate ? No need to introduce a new URI?

Comment (calvinbutcher) [inline implementations/validator/validator.py]: I didn't want to lose the GET functionality for those that already had it in place. Shall I make a commit to POST to /validate?

**Comment by azaroth42** [comment]
Merged, thanks @calvinbutcher!

**Comment by schwemmer** [comment]
Thanks, @calvinbutcher, this is very useful! I have a couple of issues/questions about this:

1.  when POSTing to `http://iiif.io/api/presentation/validator/service/validate` I get the following error: `XMLHttpRequest cannot load http://iiif.io/api/presentation/validator/service/validate. Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:3000' is therefore not allowed access. The response had HTTP status code 405.`

2. I assume the POST parameter name for the data to validate is `data`, correct?

3. Is it possible to specify the API version with a `version` parameter?

**Comment by schwemmer** [comment]
Thanks, Rob! Please let me know when it's deployed by commenting here.

**Pull request merged**